Dynavity/Dynavity/view/ToolbarView.swift

```diff
@@ -84,8 +105,112 @@ struct ToolbarView: View {
         }
     }
 
+    private var annotationMenu: some View {
+        HStack {
+            LazyVGrid(columns: columns, spacing: 20) {
+                ForEach(AnnotationPalette.annotationWidths, id: \.self) { width in
+                    Button(action: {
+                        viewModel.switchAnnotationWidth(width)
+                        shouldShowAnnotationMenu = false
+                    }) {
+                        ZStack {
+                            Circle()
+                                .fill(Color.black)
+                            Circle()
+                                .fill(Color.white)
+                                .frame(width: width, height: width, alignment: .center)
+                        }
+                    }.frame(width: selectorSize, height: selectorSize)
+                }
+            }
+            Spacer()
+            LazyVGrid(columns: columns, spacing: 20) {
+                ForEach(AnnotationPalette.annotationColors, id: \.self) { color in
+                    Button(action: {
+                        viewModel.switchAnnotationColor(color)
+                        shouldShowAnnotationMenu = false
+                    }) {
+                        Circle().fill(Color(color))
+                    }.frame(width: selectorSize, height: selectorSize)
+                }
+            }
+        }
+        .offset(x: 0, y: 80)
+        .frame(width: 200, height: 100, alignment: .center)
+    }
+
+    private var penSelectionButton: some View {
+        Button(action: {
+            if currentlySelectedTool == .pen {
+                shouldShowAnnotationMenu = true
+            }
+            currentlySelectedTool = .pen
+            viewModel.switchAnnotationTool(PKInkingTool(.pen, color: .blue, width: 10))
```

Comment:
What do you think of defining it in AnnotationPalette in the model, and using a getter from the viewModel? I feel like the default definitions don't really belong in the viewModel?

Reply:
Yep I think it sounds good to leave the definitions in the model. But maybe instead of a getter, the viewModel can have an overloaded method / another method that selects the default from the model?

Reply:
Alright! Sounds good. Not too sure if you meant a generic method that selects default from any model, or just for the annotationTool. Nonetheless I've made some updates! Will change again if necessary:)